auto_update.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/inconshreveable/go-update"
+)
+
+const locallyBuilt = "(Locally Built)"
+
+// RunWithUpdateCheck checks if an update is due, checks if current version is outdated and performs update if needed
+func (c *TGFConfig) RunWithUpdateCheck() int {
+	app := c.tgf
+	const autoUpdateFile = "TGFAutoUpdate"
+
+	if !app.AutoUpdateSet { // Fallback to config settings if --update flag is not set
+		app.AutoUpdateSet = c.AutoUpdate != ""
+		app.AutoUpdate = c.AutoUpdate == "on"
```

Comment:
Try to avoid config string such as this one that can accept any value. This is clearly a boolean, so, it must be configured as a boolean. In this case, is should be a `*bool`.

Another simpler way is to use 0 as the update delay. So if AutoUpdateDelay is set to 0, the auto-update is disabled.

Reply:
I think it should be a bool. It is more self documenting